src/t_list.c

```diff
@@ -630,9 +641,9 @@ void lsetCommand(client *c) {
         notifyKeyspaceEvent(NOTIFY_LIST,"lset",c->argv[1],c->db->id);
         server.dirty++;
 
-        /* We might replace a big item with a small one or vice versa,
+        /* We might replace a big item with a small one,
```

Comment:
maybe add a comment that the above call to `listTypeTryConversionForValues` handled the other case?

Reply:
while writing unit tests, I found that the conversion in the LSET command may not work well.
The main reason is that `quicklistReplaceEntry` does not merge nodes after replacement, which means it does not satisfy the need to convert to a listpack (only one node).
The only case where this works is when the quicklist has only one node and has not yet reached the threshold when the `LSET` command is used to replace it after reaching the threshold.
One alternative would be to merge after the quicklist replacement, but I'm not sure if that would cause a performance decrement.